### PR TITLE
adding start files

### PR DESCRIPTION
**Pull request by rshade**

### CODE REVIEW

**Review by cdwilhelm — COMMENTED** [review]
Is this a chef file of puppet file.  name should probably be changed and file name too.

**Review by cdwilhelm — COMMENTED** [review]
what happens when this runs on non-aws clouds?

File: puppet-client/nd_backup/06b_puppet_config.sh [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+PRIMARY_NIC=$(route | grep default | awk '{print $8}') # eth0 or ens3
+MAC=${MAC:-$(cat /sys/class/net/${PRIMARY_NIC}/address)}
+
+# These facts can be overridden in RightScale - but we default to using the
+# Amazon Metadata service to get the info. This will allow us in the future to
+# pass in options to this script for different clouds.. but its not necessary
+# for now.
+IMAGE_NAME=${IMAGE_NAME:-$(curl -s http://169.254.169.254/latest/meta-data/ami-id)}
+INSTANCE_ID=${INSTANCE_ID:-$(curl -s http://169.254.169.254/latest/meta-data/instance-id)}
+ZONE=${ZONE:-$(curl -s http://169.254.169.254/latest/meta-data/placement/availability-zone)}
```
Comment (cdwilhelm) [inline puppet-client/nd_backup/06b_puppet_config.sh]: what happens when this runs on non-aws clouds?

File: puppet-client/nd_backup/Chef_Client_Delete_Node.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#! /usr/bin/sudo /bin/bash
+# ---
+# RightScript Name: Chef Client Delete Node
```
Comment (cdwilhelm) [inline puppet-client/nd_backup/Chef_Client_Delete_Node.sh]: Is this Chef or puppet?  name and file should be changed to reflect puppet

File: puppet-client/nd_backup/Chef_Client_Delete_Node.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# Description: Remove the node from the Chef Server.  Run this as a Decommission Script
```
Comment (cdwilhelm) [inline puppet-client/nd_backup/Chef_Client_Delete_Node.sh]: Chef?

File: puppet-client/nd_backup/Chef_Client_Delete_Node.sh [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+#     Possible Values:
+#     - text:true
+#     - text:false
+# Attachments: []
+# ...
+set -e
+
+if [ "$DECOM_REASON" == "terminate" ] && [ "$DELETE_NODE_ON_TERMINATE" == "true" ];then
+  echo "Removing node ${HOSTNAME} from chef server"
+  knife node delete --yes -c /etc/chef/client.rb ${HOSTNAME}
```
Comment (cdwilhelm) [inline puppet-client/nd_backup/Chef_Client_Delete_Node.sh]: not chef

File: puppet-client/nd_backup/Chef_Client_Run_Recipe.sh [hunk 5]
```diff
@@ -0,0 +1,56 @@
+#! /usr/bin/sudo /bin/bash
+# ---
+# RightScript Name: Chef Client Run Recipe
```
Comment (cdwilhelm) [inline puppet-client/nd_backup/Chef_Client_Run_Recipe.sh]: chef?

File: puppet-client/nd_backup/userdata.sh [hunk 6]
```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+set -euo pipefail
+
+#set aws settings
+declare -x PP_INSTANCE_ID=$(curl --silent --show-error --retry 3 http://169.254.169.254/latest/meta-data/instance-id)
```
Comment (cdwilhelm) [inline puppet-client/nd_backup/userdata.sh]: what happens when this runs on none-aws clouds

**Review by cdwilhelm — COMMENTED** [review]
what is the nd_backup directory about. is that needed, or can it be collapsed so all the files are in puppet-client.    I think any file that is not used now should be removed until it's needed in the future

File: puppet-client/nd_backup/06b_puppet_config.sh (hunk 1 above)
Comment (rshade) [reply puppet-client/nd_backup/06b_puppet_config.sh]: All of the stuff in nd_backup is just for reference. I am unsure how people will want to use it, and I don't want to lose the code until we are sure it works in their use-cases. The only thing running is the stuff defined in the server_template.yml

**Review by cdwilhelm — APPROVED** [review]
👍

**Pull request merged**